### ISSUE TITLE
Space/time folds are equal for test set but not training set when comparing sptcv_cstf vs CAST::CreateSpacetimeFolds

### ISSUE THREAD

**Issue by KennethTM** [mlr-org/mlr3spatiotempcv#134]
Hi 

I came by this magnificent package while experimenting with the mlr3 package, and it seems just to be what I was looking for - thanks!

However, I have wondered why my training sets contained fewer observations than expected when using the "sptcv_cstf" resampling. 

When running a test from your codebase, it seems that results differ for test and training set when using sptcv_cstf vs CAST::CreateSpacetimeFolds. 

Is this a bug or have I missed something?

Best,
Kenneth

Reproducing my observation:

```r
library(mlr3);library(mlr3spatiotempcv);library(CAST)

#sptcv_cstf vs. CAST::CreateSpacetimeFolds on test set:
task = tsk("cookfarm")
rsp = rsmp("sptcv_cstf", folds = 10, time_var = "Date", space_var = "SOURCEID")
set.seed(42)
rsp$instantiate(task)
rsp$test_set(1)

set.seed(42)
indices = CAST::CreateSpacetimeFolds(task$data(), "SOURCEID", "Date")
indices$indexOut[[1]]

#As expected, indexes for test set are the same

#sptcv_cstf vs. CAST::CreateSpacetimeFolds on train set:
task = tsk("cookfarm")
rsp = rsmp("sptcv_cstf", folds = 10, time_var = "Date", space_var = "SOURCEID")
set.seed(42)
rsp$instantiate(task)
rsp$train_set(1)

set.seed(42)
indices = CAST::CreateSpacetimeFolds(task$data(), "SOURCEID", "Date")
indices$index[[1]]

#Not what I expected, indexes for training set differ

```

**Comment by pat-s** [comment]
Yes, this is a bug - caused by a missing `!`.

And for whatever reason I only checked the test sets to match, not the train sets.

I'll push a fix to CRAN within the next days, until then you can use the GH version. 

Thanks for reporting!

**Comment by KennethTM** [comment]
Awesome - thanks for the quick fix!
